packages/sdk/src/services/tokenBalance.ts

```diff
@@ -77,45 +76,51 @@ export async function fetchERC20Balance(
   )
 
   return {
-    symbol: token.symbol,
+    symbol: asset.metadata.symbol,
     totalBalance,
-    totalFormattedBalance: formatUnits(totalBalance, token.decimals),
+    totalFormattedBalance: formatUnits(totalBalance, asset.metadata.decimals),
     chainBalances,
   }
 }
 
 /**
- * Fetch balance for this token on a specific chain
+ * Fetch balance for this asset on a specific chain
  */
 async function fetchERC20BalanceForChain(
-  token: TokenInfo,
+  asset: Asset,
   chainId: SupportedChainId,
   walletAddress: Address,
   chainManager: ChainManager,
 ): Promise<{ balance: bigint; tokenAddress: Address }> {
-  const tokenAddress = getTokenAddress(token.symbol, chainId)
+  const tokenAddress = asset.address[chainId]
   if (!tokenAddress) {
-    throw new Error(`${token.symbol} not supported on chain ${chainId}`)
+    throw new Error(
+      `${asset.metadata.symbol} not supported on chain ${chainId}`,
+    )
   }
 
   const publicClient = chainManager.getPublicClient(chainId)
 
   // Handle native ETH balance
-  if (token.symbol === 'ETH') {
+  if (asset.type === 'native') {
     return {
       balance: await publicClient.getBalance({
         address: walletAddress,
       }),
-      tokenAddress,
+      tokenAddress: '0x0000000000000000000000000000000000000000' as Address,
```

Comment:
nit: isnt this address already defined in the `ETH` token? or is there a reason we have to hardcode this?